app/models/ipeds_ic_py.rb

```diff
@@ -0,0 +1,142 @@
+    'xciptui3' => { column: :xciptui3, converter: BaseConverter },
+    'ciptuit3' => { column: :ciptuit3, converter: BaseConverter },
+    'xcipsup3' => { column: :xcipsup3, converter: BaseConverter },
+    'cipsupp3' => { column: :cipsupp3, converter: BaseConverter },
+    'xciplgt3' => { column: :xciplgt3, converter: BaseConverter },
+    'ciplgth3' => { column: :ciplgth3, converter: BaseConverter },
+    'prgmsr3' => { column: :prgmsr3, converter: BaseConverter },
+    'xmthcmp3' => { column: :xmthcmp3, converter: BaseConverter },
+    'mthcmp3' => { column: :mthcmp3, converter: BaseConverter },
+    'cipcode4' => { column: :cipcode4, converter: BaseConverter },
+    'xciptui4' => { column: :xciptui4, converter: BaseConverter },
+    'ciptuit4' => { column: :ciptuit4, converter: BaseConverter },
+    'xcipsup4' => { column: :xcipsup4, converter: BaseConverter },
+    'cipsupp4' => { column: :cipsupp4, converter: BaseConverter },
+    'xciplgt4' => { column: :xciplgt4, converter: BaseConverter },
+    'ciplgth4' => { column: :ciplgth4, converter: BaseConverter },
+    'prgmsr4' => { column: :prgmsr4, converter: BaseConverter },
+    'xmthcmp4' => { column: :xmthcmp4, converter: BaseConverter },
+    'mthcmp4' => { column: :mthcmp4, converter: BaseConverter },
+    'cipcode5' => { column: :cipcode5, converter: BaseConverter },
+    'xciptui5' => { column: :xciptui5, converter: BaseConverter },
+    'ciptuit5' => { column: :ciptuit5, converter: BaseConverter },
+    'xcipsup5' => { column: :xcipsup5, converter: BaseConverter },
+    'cipsupp5' => { column: :cipsupp5, converter: BaseConverter },
+    'xciplgt5' => { column: :xciplgt5, converter: BaseConverter },
+    'ciplgth5' => { column: :ciplgth5, converter: BaseConverter },
+    'prgmsr5' => { column: :prgmsr5, converter: BaseConverter },
+    'xmthcmp5' => { column: :xmthcmp5, converter: BaseConverter },
+    'mthcmp5' => { column: :mthcmp5, converter: BaseConverter },
+    'cipcode6' => { column: :cipcode6, converter: BaseConverter },
+    'xciptui6' => { column: :xciptui6, converter: BaseConverter },
+    'ciptuit6' => { column: :ciptuit6, converter: BaseConverter },
+    'xcipsup6' => { column: :xcipsup6, converter: BaseConverter },
+    'cipsupp6' => { column: :cipsupp6, converter: BaseConverter },
+    'xciplgt6' => { column: :xciplgt6, converter: BaseConverter },
+    'ciplgth6' => { column: :ciplgth6, converter: BaseConverter },
+    'prgmsr6' => { column: :prgmsr6, converter: BaseConverter },
+    'xmthcmp6' => { column: :xmthcmp6, converter: BaseConverter },
+    'mthcmp6' => { column: :mthcmp6, converter: BaseConverter }
+  }.freeze
+
+  validates :cross, presence: true
+  validates :chg1py3, numericality: true, allow_blank: true
+  validates :books, numericality: true, allow_blank: true
+
+  before_validation :derive_dependent_columns
+
+  def derive_dependent_columns
+    self.tuition_in_state = chg1py3
+    self.tuition_out_of_state = chg1py3
```

Comment:
Are we setting a class variable based on an instance variable? Why do we need to do "self." here, isn't self the instance? I think you can just do "tuition_in_state = " without the self